x/or/da/keeper/cc_test.go

```diff
@@ -0,0 +1,508 @@
+package keeper_test
+
+import (
+	"bytes"
+	"compress/zlib"
+	"time"
+
+	"github.com/klauspost/compress/zstd"
+
+	sdktypes "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/x/or/da/types"
+)
+
+func (s *KeeperTestSuite) TestDecompressCCBatch() {
+	src := types.CCBatch{
+		ShouldStartAtFrame: 30,
+		Frames: []*types.CCBatchFrame{
+			{
+				Header: &types.CCBatchHeader{
+					ParentHash: []byte("parent_hash"),
+					Timestamp:  time.Now().UTC(),
+					L2Height:   10,
+					L1Height:   100,
+				},
+				Elements: []*types.CCBatchElement{
+					{
+						Txraw:      []byte("Transactions"),
+						QueueIndex: 10,
+					},
+				},
+			},
+		},
+	}
+
+	serializedSrc := s.encCfg.Marshaler.MustMarshal(&src)
+
+	testCases := map[string]struct {
+		compressor func(src []byte) []byte
+		opt        types.CompressionOption
+		err        bool
+	}{
+		"success - zlib": {
+			compressor: func(src []byte) []byte {
+				var b bytes.Buffer
+				w := zlib.NewWriter(&b)
+				_, err := w.Write(serializedSrc)
+				s.Require().NoError(err)
+				err = w.Close()
+				s.Require().NoError(err)
+				return b.Bytes()
+			},
+			opt: types.OptionZLIB,
+			err: false,
+		},
+		"wrong compressed - zstd": {
+			compressor: func(src []byte) []byte {
+				var b bytes.Buffer
+				w, _ := zstd.NewWriter(&b)
+				_, err := w.Write(serializedSrc)
+				s.Require().NoError(err)
+				err = w.Close()
+				s.Require().NoError(err)
+				return b.Bytes()
+			},
+			opt: types.OptionZSTD,
+			err: true,
+		},
+		"option empty": {
+			compressor: nil,
+			opt:        types.OptionEmpty,
+			err:        true,
+		},
```

Comment:
Would you add a TestCase that raises ErrInvalidCompressedData?

Reply:
second, third case generate that error

Reply:
Certainly, the second and third cases also generate ErrInvalidCompressedData.
What I meant was would you add a test case where `uint64(len(origin.Data)) > p.CCBatchMaxBytes` is true.